keras_cv/visualization/plot_segmentation_mask_gallery.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from keras_cv import utils
+from keras_cv.utils import assert_matplotlib_installed
+from keras_cv.visualization.plot_image_gallery import plot_image_gallery
+
+
+def build_compatile_segmentation_masks(segmentation_masks):
+    rank = len(segmentation_masks.shape)
+    if rank == 3:
+        # (B, H, W)
+        return segmentation_masks[..., np.newaxis]
+    elif rank == 4:
+        # (B, H, W, num_channels) OR (B, H, W, 1)
+        if segmentation_masks.shape[-1] == 1:
+            return segmentation_masks.repeat(repeats=3, axis=-1)
```

Comment:
This `3` is for RGB, right?

That's worth a code comment because it's a bit confusing as a magic constant